packages/jest-matchers/src/index.js

```diff
@@ -10,27 +10,40 @@
 
 'use strict';
 
-import type {ThrowingMatcherFn, ExpectationResult} from '../types';
+import type {
+  Expect,
+  ExpectationResult,
+  ExpectationObject,
+  MatchersObject,
+  RawMatcherFn,
+  ThrowingMatcherFn,
+} from '../types';
 
 const matchers = require('./matchers');
+const GLOBAL_MATCHERS_OBJECT_SYMBOL = Symbol.for('$$jest-matchers-object');
 
-function expect(actual: any) {
+if (!global[GLOBAL_MATCHERS_OBJECT_SYMBOL]) {
+  Object.defineProperty(global, GLOBAL_MATCHERS_OBJECT_SYMBOL, {value: {}});
```

Comment:
It should use `Object.create(null)` as value. We don't want the prototype chain as part of the matchers.

I can live with this, even though I'd prefer it not to be global but I take the module registry reset argument. I'm happy to change this later when we inject `jest.addMatchers` properly.